server/src/main/java/org/elasticsearch/action/support/master/AcknowledgedResponse.java

```diff
@@ -77,6 +77,18 @@ protected void writeAcknowledged(StreamOutput out) throws IOException {
         out.writeBoolean(acknowledged);
     }
 
+    @Override
+    public void readFrom(StreamInput in) throws IOException {
+        super.readFrom(in);
+        readAcknowledged(in);
```

Comment:
I am on mobile so I can not look this up. Is there a reason these read/write methods need to be visible to sub-classes now with this change?

Reply:
Good point. Unfortunately there are (only) six classes extending AcknowledgedResponse that use those directly at the moment to read/write the acknowledged flag. I just looked at the details and found a problem with bwc with this change because they also use super() and interleave reading/writing the acknowledged flag in s weird way. I'm down to three cases where it might make sense to change the order of serialization in a bwc way to make this change work.